modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -166,7 +167,7 @@ def grab_nas
     if /[\x00]{16,}(?<ftpuser>[[:print:]]+)[\x00]{16,}(?<ftppass>[[:print:]]+)/ =~ data
       ftpuser.strip!
       ftppass.strip!
-      unless ftpuser.blank? || ftppass.blank?
+      if !ftpuser.blank? || !ftppass.blank?
```

Comment:
Actually yes it probably should, or I could just move it back to unless and remove the not empty statements.